src/compiler/transformers/esnext.ts

```diff
@@ -16,9 +19,67 @@ namespace ts {
                 return node;
             }
             switch (node.kind) {
+                case SyntaxKind.BinaryExpression:
+                    const binaryExpression = <BinaryExpression>node;
+                    if (isLogicalOrCoalescingAssignmentOperator(binaryExpression.operatorToken.kind)) {
+                        return transformLogicalAssignmentOperators(binaryExpression);
+                    }
+                    // falls through
                 default:
                     return visitEachChild(node, visitor, context);
             }
         }
+
+        function transformLogicalAssignmentOperators(binaryExpression: BinaryExpression): VisitResult<Node> {

```

Comment:
nit: Use the singular for "Operator" here (or remove it altogether):
```suggestion
        function transformLogicalAssignmentOperator(binaryExpression: BinaryExpression): VisitResult<Node> {
```
```suggestion
        function transformLogicalAssignment(binaryExpression: BinaryExpression): VisitResult<Node> {
```